prow/tide/tide.go

```diff
@@ -690,15 +690,15 @@ func (c *Controller) mergePRs(sp subpool, prs []PullRequest) error {
 	return nil
 }
 
-func (c *Controller) trigger(sp subpool, presubmits map[int]sets.String, prs []PullRequest) error {
-	requiredJobs := sets.NewString()
+func (c *Controller) trigger(sp subpool, presubmitContexts map[int]sets.String, prs []PullRequest) error {
+	requiredContexts := sets.NewString()
 	for _, pr := range prs {
-		requiredJobs = requiredJobs.Union(presubmits[int(pr.Number)])
+		requiredContexts = requiredContexts.Union(presubmitContexts[int(pr.Number)])
 	}
 
 	// TODO(cjwagner): DRY this out when generalizing triggering code (and code to determine required and to-run jobs).
```

Comment:
We need to do this and hide some of this detail behind some abstraction that will make it that much harder to make this mistake in the future. Lots of stringly-typed fields here right now and we should leverage the type system to make our lives easier.